src/ripple/app/ledger/InboundLedger.h

```diff
@@ -65,10 +65,10 @@ class InboundLedger
     // Called when another attempt is made to fetch this same ledger
     void update (std::uint32_t seq);
 
-    std::shared_ptr<Ledger> const&
+    std::shared_ptr<Ledger const>
     getLedger() const
     {
-        return mLedger;
+        return std::const_pointer_cast<Ledger const> (mLedger);
```

Comment:
The `const_pointer_cast` is not needed here. A `std::shared_ptr<T>` implicitly converts to `std::shared_ptr<T const>` via [constructor 9](http://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr).